pkg/cmd/connector/update/update.go

```diff
@@ -35,85 +35,131 @@ func NewUpdateCommand(f *factory.Factory) *cobra.Command {
 		Short:   f.Localizer.MustLocalize("connector.update.cmd.shortDescription"),
 		Long:    f.Localizer.MustLocalize("connector.update.cmd.longDescription"),
 		Example: f.Localizer.MustLocalize("connector.update.cmd.example"),
-		Hidden:  true,
 		Args:    cobra.NoArgs,
+		Hidden:  true,
 		RunE: func(cmd *cobra.Command, args []string) error {
-
 			validOutputFormats := flagutil.ValidOutputFormats
 			if opts.outputFormat != "" && !flagutil.IsValidInput(opts.outputFormat, validOutputFormats...) {
 				return flagutil.InvalidValueError("output", opts.outputFormat, validOutputFormats...)
 			}
 
-			return runUpdateCommand(opts)
+			return runUpdate(opts)
 		},
 	}
-
-	flags := connectorcmdutil.NewFlagSet(cmd, f)
-	flags.StringVar(&opts.file, "file", "", f.Localizer.MustLocalize("connector.file.flag.description"))
-	flags.AddConnectorID(&opts.id)
+	flags := flagutil.NewFlagSet(cmd, f.Localizer)
+	flags.StringVar(&opts.name, "name", "", f.Localizer.MustLocalize("connector.flag.name.description"))
+	flags.StringVar(&opts.namespaceID, "namespace-id", "", f.Localizer.MustLocalize("connector.flag.kafkaID.description"))
+	flags.StringVar(&opts.kafkaID, "kafka-id", "", f.Localizer.MustLocalize("connector.flag.namespaceID.description"))
 	flags.AddOutput(&opts.outputFormat)
 
 	return cmd
-
 }
 
-func runUpdateCommand(opts *options) error {
-	f := opts.f
+func runUpdate(opts *options) error {
 
-	var conn connection.Connection
-	conn, err := f.Connection()
+	conn, err := opts.f.Connection()
 	if err != nil {
 		return err
 	}
 
-	var specifiedFile *os.File
-	if opts.file == "" {
-		file, err1 := util.CreateFileFromStdin()
-		if err1 != nil {
+	api := conn.API()
+
+	connector, err := contextutil.GetCurrentConnectorInstance(&conn, opts.f)
+	if err != nil || connector == nil {
+		if connector, err = connectorutil.InteractiveSelect(conn, opts.f); err != nil {
 			return err
 		}
-		specifiedFile = file
-	} else {
-		if util.IsURL(opts.file) {
-			specifiedFile, err = util.GetContentFromFileURL(f.Context, opts.file)
-		} else {
-			specifiedFile, err = os.Open(opts.file)
-		}
+	}
+
+	connectorChanged := false
+	if opts.name != "" {
+		connector.SetName(opts.name)
+		connectorChanged = true
+	}
+	if opts.namespaceID != "" {
+		connector.SetNamespaceId(opts.namespaceID)
+		connectorChanged = true
+	}
+	if opts.kafkaID != "" {
+		connector.Kafka.SetId(opts.kafkaID)
+		connectorChanged = true
+	}
+
+	if !connectorChanged {
+		err = runEditor(connector)
```

Comment:
I forgot to put some message to user informing them that now we will start editor for them and they 
can change editor by changing $EDITOR env. Like EDITOR="Code -w"